vulcanai2/models/basenetwork.py

```diff
@@ -99,50 +103,50 @@ def __init__(self, name, dimensions, config, save_path=None, input_networks=None
             validation_accuracy=[]
         )
 
-        if self._dimensions:
-            if len(self._dimensions)==1: # Check if length of self._dimensions list is 1, then self.in_dim is the first tuple of self._dimensions
-                self.in_dim = self._dimensions[0]
-            else:
-                # if len(self._dimensions) > 1, then iterate and concatenate all dims
-                self.in_dim = []
-                for net in self._input_networks:
-                    self.in_dim.append(net.in_dim)
-        #import pdb; pdb.set_trace()
-
         self._create_network(
             activation=activation,
             pred_activation=pred_activation)
-       
-        out_shapes = self.get_output_shapes()
-        self.out_dim = out_shapes[list(out_shapes)[-1]]['output_shape'][1:] 
-
-    # TODO: ignore for now
-    # def get_in_dim(self, network = None):
-    #     if not network:
-    #         network = self
-    #     for net in network.input_networks:
-    #             if net.input_networks is None:
-                    
-    #                 self.in_dim.append(net.get_output_shapes())
-
-    #             else:
-    #                 self.in_dim = self.get_in_dim(net)
-    #             temp_tensor = torch.zeros([1, *net.in_dim])
+
+        #print(self)
+        
+        out_shapes = self.get_output_shapes(network=self.network,input_size=self.in_dim)
```

Comment:
if it's already a class function, you shouldn't need to pass it a self object

Reply:
I'm overriding it to just do a forward pass on self.network alone. Check here: https://github.com/Aifred-Health/Vulcan2/blob/update_branch2/vulcanai2/models/basenetwork.py#L310-L320

Reply:
I see. we will need to come back to get_output_shapes to clean it up.